packages/graphql-typescript-definitions/src/index.ts

```diff
@@ -40,22 +49,48 @@ export interface Build {
   fragments: Fragment[];
 }
 
+// projectName => { filePath => document }
+export type GraphQLProjectDocuments = Map<string, Map<string, DocumentNode>>;
+
 export class Builder extends EventEmitter {
   watching = false;
-  private globs: string;
-  private schemaPath: string;
-  private schema!: GraphQLSchema;
-  private options: Pick<
-    Options,
-    Exclude<keyof Options, 'graphQLFiles' | 'schemaPath'>
-  >;
+  private options: Options;
+  // workspace graphql configuration
+  // see: https://github.com/prisma/graphql-config
+  private readonly config: GraphQLConfig;
   private documentCache = new Map<string, DocumentNode>();
 
-  constructor({graphQLFiles, schemaPath, ...options}: Options) {
+  constructor({
+    cwd,
+    graphQLFiles,
+    schemaPath,
+    schemaTypesPath,
+    ...options
+  }: Options & Partial<CLIOptions>) {
     super();
-    this.globs = graphQLFiles;
-    this.schemaPath = schemaPath;
     this.options = options;
+
+    if (cwd && graphQLFiles && schemaPath && schemaTypesPath) {
+      this.config = new GraphQLConfig(
+        {
+          // graphql-config typings are broken so we have to put a dummy schemaPath here
+          schemaPath: '',
+          projects: {
+            cli: {
+              schemaPath,
+              includes: [graphQLFiles],
+              extensions: {
+                schemaTypesPath,
+              },
+            },
+          },
+        },
+        // graphql-config depends on the location of this file so it needs to be sane
+        resolve(cwd, '.graphqlconfig'),
+      );
+    } else {
+      this.config = getGraphQLConfig();
```

Comment:
I should probably be checking `graphQLFiles` here to see if it is a `*.graphqlconfig` path and then construct the config accordingly.